### PR TITLE
lib: dk_buttons_and_leds: fix for nRF9160DK IO expander

### PR DESCRIPTION
**Pull request by eivindj-nordic**
The driver for the PCAL6408A IO expander on the nRF9160DK does not support disabling the interrupts while they are being processed. Move disabling of the interrupts to the buttons_scan_fn function. Also, the IO expander driver only supports edge driven interrupts, so this is added to the configuration.

### CODE REVIEW

**Review by divipillai — APPROVED** [review]

**Review by kapi-no — CHANGES_REQUESTED** [review]
@eivindj-nordic, you seem to pull in the manifest change for Find My by accident. Please remove the following commit:

e75963b52486f653e6da536951e0c3fafcb76a71

**Comment by eivindj-nordic** [comment]
> @eivindj-nordic, you seem to pull in the manifest change for Find My by accident. Please remove the following commit:

Yes, it is only temporary to see if that fixes the issues with failures on the `test-sdk-find-my` plan. I will remove it/wait for it to get merged before merging this one.

**Comment by kapi-no** [comment]
@miha-nordic, @rawoNCS, please take a look why the Find My CI is failing

**Comment by eivindj-nordic** [comment]
> Looks like environmental issue with access to artifactory.

@miha-nordic Do you know what is the status on this issue? If I understand you correctly it should not be related to this PR in this case(?). I see that this test is using the buttons, which this PR is changing the handling of. Could it be related? If so, how is the buttons used in CI for this? Is there any way I can reproduce locally?

**Comment by eivindj-nordic** [comment]
> @eivindj-nordic, you seem to pull in the manifest change for Find My by accident. Please remove the following commit:
> 
> [e75963b](https://github.com/nrfconnect/sdk-nrf/commit/e75963b52486f653e6da536951e0c3fafcb76a71)

@kapi-no The manifest change commit is removed.

**Comment by miha-nordic** [comment]
> > Looks like environmental issue with access to artifactory.
> 
> @miha-nordic Do you know what is the status on this issue? If I understand you correctly it should not be related to this PR in this case(?). I see that this test is using the buttons, which this PR is changing the handling of. Could it be related? If so, how is the buttons used in CI for this? Is there any way I can reproduce locally?

Access to artifactory is not a problem anymore. Issue that occurs here could be related to button handling. Case that is not working is pressing button for 3s and log should be printed on console. You can reproduce it bu flashing device with Find-my qualification sample and manually hold button. in CI we use nrf52840 to send button press signals.

**Review by jhn-nordic — APPROVED** [review]

**Review by kapi-no — APPROVED** [review]
Approving as this PR does not change Find My component anymore

File: lib/dk_buttons_and_leds/dk_buttons_and_leds.c
Comment (MirkoCovizzi) [inline lib/dk_buttons_and_leds/dk_buttons_and_leds.c:115]: Nit: let's keep the pattern of 'err' and 'if (err)' with no empty line in between

**Review by MirkoCovizzi — APPROVED** [review]

**Pull request merged**
